pkg/pillar/types/attesttypes.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	uuid "github.com/satori/go.uuid"
+)
+
+//AttestNonce carries nonce published by requester
+type AttestNonce struct {
+	nonce     []byte
+	requester string
+}
+
+//Key returns nonce content, which is the key as well
+func (nonce AttestNonce) Key() string {
+	return nonce.requester
+}
+
+//SigAlg denotes the Signature algorithm in use e.g. ECDSA, RSASSA
+type SigAlg uint8
+
+//CertType carries the certificate use case e.g. ek, ecdh_exchange etc
+type CertType uint8
+
+//Various certificate types published by tpmmgr
+const (
+	EcdsaSha256 SigAlg = iota + 0 // Default
```

Comment:
In other places we've defined a FooNone for 0, and the actual values starting with 1. I don't think that is strictly required unless there is code somewhere which wants to ask "is Foo set?"